lib/ruby_indexer/lib/ruby_indexer/index.rb

```diff
@@ -380,38 +379,36 @@ def index_all(uris: @configuration.indexable_uris, &block)
 
     sig { params(uri: URI::Generic, source: T.nilable(String), collect_comments: T::Boolean).void }
     def index_single(uri, source = nil, collect_comments: true)
-      full_path = uri.full_path
-      return unless full_path
+      content = case uri.scheme
+      when "file"
+        full_path = uri.full_path
+        source || ((full_path = uri.full_path) && File.read(full_path))
+      when "untitled"
+        source
+      end
+
+      unless content
```

Comment:
Could this move to before the `when`?

Reply:
I refactored this method to require the source to be passed in, so that we don't have to handle this here.